app/src/main/java/pm/gnosis/heimdall/data/db/ERC20TokenDao.kt

```diff
@@ -18,6 +15,6 @@ interface ERC20TokenDao {
     @Query("SELECT * FROM ${ERC20TokenDb.TABLE_NAME} ORDER BY ${ERC20TokenDb.COL_NAME} ASC")
     fun observeTokens(): Flowable<List<ERC20TokenDb>>
 
-    @Query("DELETE FROM ${ERC20TokenDb.TABLE_NAME} WHERE ${ERC20TokenDb.COL_ADDRESS} = :address")
-    fun deleteToken(address: String)
+    @Delete
+    fun deleteToken(erC20Token: ERC20TokenDb)
```

Comment:
to be honest I am not a big fan of this delete, because we always need to create an instance of the DB model and use that for deletion. That was also the reason why I replaced it with the deletion by address in the first place.